sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -2019,17 +2028,10 @@ public void testCountNullableExpression() throws Exception
                   .dataSource(CalciteTests.DATASOURCE1)
                   .intervals(querySegmentSpec(Filtration.eternity()))
                   .granularity(Granularities.ALL)
-                  .virtualColumns(
-                      expressionVirtualColumn(
-                          "v0",
-                          "case_searched((\"dim2\" == 'abc'),'yes',(\"dim2\" == 'def'),'yes'," + DruidExpression.nullLiteral() + ")",
-                          ValueType.STRING
-                      )
-                  )
                   .aggregators(aggregators(
                       new FilteredAggregatorFactory(
                           new CountAggregatorFactory("a0"),
-                          not(selector("v0", NullHandling.defaultStringValue(), null))
+                          new InDimFilter("dim2", ImmutableList.of("abc", "def"), null)
```

Comment:
Here we have `COUNT(CASE WHEN dim2 = 'abc' THEN 'yes' WHEN dim2 = 'def' THEN 'yes' END)` being translated as,

- Old: `COUNT(*) FILTER(WHERE CASE WHEN dim2 = 'abc' THEN 'yes' WHEN dim2 = 'def' THEN 'yes' ELSE NULL END IS NOT NULL`
- New: `COUNT(*) FILTER(WHERE dim2 IN ('abc', def'))`

The new translation is equivalent, and definitely cleaner.